### PR TITLE
Cypress general updates

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

- update dependencies
- update `docker` image
- move `themes.test.js` into `carbon-provider.test.js`

### Current behaviour

- an old version of cypress
- an old version of `chrome/docker` image

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] All tests should pass

### CODE REVIEW

**Review by ZhuoyuJin — APPROVED** [review]

File: src/components/carbon-provider/carbon-provider.test.js [hunk 1]
```diff
@@ -1,11 +1,275 @@
...
+  "rgb(179, 217, 200)",
+];
+
 context("Testing Carbon Provider component", () => {
-  describe.skip("should render Carbon Provider component", () => {
-    // functionality of theming is tested in `theme.test.js` file
-  });
+  describe.each(buildTestArray(commonColors))(
+    "should render components with %s theme",
+    (theme, themeName, color) => {
```
Comment (Parsium) [inline src/components/carbon-provider/carbon-provider.test.js:77]: Given the definition of `buildTestArray` on line 34, I think these parameter names need swapping around

```suggestion
    (themeName, theme, color) => {
```

Comment (DlgSHi) [reply src/components/carbon-provider/carbon-provider.test.js:77]: I don't think it matters :)

Comment (DlgSHi) [reply src/components/carbon-provider/carbon-provider.test.js:77]: it is used properly :)

Comment (Parsium) [reply src/components/carbon-provider/carbon-provider.test.js:77]: The tests still work as intended, however it is a bit misleading, as `themeName` could be read as a string when it's actually a theme object.

**Review by Parsium — APPROVED** [review]

**Pull request merged**